File: TabCommon/Library/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/base/BaseApiCreator.java
package com.nextzy.library.base.mvvm.layer3Repository.network.base;


import okhttp3.OkHttpClient;
import th.co.thekhaeng.waterlibrary.java.base.nextwork.engine.NextworkApiCreator;

/**
 * Created by thekhaeng on 5/10/2017 AD.
 */

public abstract class BaseApiCreator<T> extends NextworkApiCreator<T>{
    public static final String BASE_URL = "";

    private static BaseApiCreator instance;

    public BaseApiCreator(Class<T> apiClass) {
        super(apiClass);
    }

    @Override
    public String getBaseUrl() {
        return BASE_URL;
    }

    @Override
    public OkHttpClient getClient(){
        return DefaultClient.getInstance().getOkHttpClient();
    }
}

File: TabCommon/CommonLibrary/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/base/BaseApiCreator.kt
package com.nextzy.library.base.mvvm.layer3Repository.network.base


import com.nextzy.nextwork.engine.NextworkApiCreator
import okhttp3.OkHttpClient

/**
 * Created by「 The Khaeng 」on 03 Oct 2017 :)
 */

abstract class BaseApiCreator<T>(apiClass: Class<T>) : NextworkApiCreator<T>(apiClass) {

    companion object {
        val BASE_URL = ""
    }

    override
    fun getBaseUrl(): String = BASE_URL

    override
    fun getClient(): OkHttpClient = DefaultClient.okHttpClient

//    override
//    fun getAdapterFactory(): CallAdapter.Factory = LiveDataCallAdapterFactory.create()
}
